test/attr/requires_and_ignores_super.swift

```diff
@@ -0,0 +1,90 @@
+// RUN: %target-swift-frontend -typecheck %s -verify
+
+///////          ///////
+///////  ERRORS  ///////
+///////          ///////
+
+// MARK: Missing 'super' call
+
+class Invalid1A {
+  @requiresSuper
+  func foo(bar: Int) {}
```

Comment:
I think we also need a more tests for error scenarios e.g. adding this attribute to a struct method, any (static or instance and/or final) property, any subscript, initializer, declaring it with an invalid message format etc.

Reply:
I'd like to see ObjC -> Swift tests for having this on `-init` and `-dealloc` as well.

Reply:
Added!